packages/blockchain-wallet-v4-frontend/src/data/goals/sagas.js

```diff
@@ -210,7 +210,10 @@ export default ({ api }) => {
     const userTiers = (yield select(
       selectors.modules.profile.getUserTiers
     )).getOrElse({})
-    if (kycNotFinished && propEq('current', 1, userTiers)) {
+    const userCanUpgrade =
+      kycNotFinished && userTiers && propEq('current', 1, userTiers)
```

Comment:
It would be simpler and easier to understand like this:

`kycNotFinished && userTiers && userTiers.current === 1`